core/src/main/scala/org/http4s/headers/Transfer-Encoding.scala

```diff
@@ -18,14 +18,18 @@ package org.http4s
 package headers
 
 import cats.data.NonEmptyList
-import cats.syntax.eq._
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.{Parser, Parser1}
+import cats.syntax.all._
+import org.http4s.internal.parsing.Rfc2616BasicRules.listSep
 
 object `Transfer-Encoding`
     extends HeaderKey.Internal[`Transfer-Encoding`]
     with HeaderKey.Recurring {
   override def parse(s: String): ParseResult[`Transfer-Encoding`] =
-    HttpHeaderParser.TRANSFER_ENCODING(s)
+    ParseResult.fromParser(parser, "Invalid Transfer-Encoding")(s)
+
+  private[http4s] val parser: Parser1[`Transfer-Encoding`] =
+    Parser.rep1Sep(TransferCoding.parser, 1, listSep).map(apply)
```

Comment:
I think this should be `Rfc7230.headerRep1`.  There are complications beyond the comma in the newer standard.